### PR TITLE
set instance field when creating new group objects

### PR DESCRIPTION
**Pull request by hppritcha**
related to #10524 and #10429

Also related to the isolation concept for sessions (instances in open mpi internal parlance).
See section 11.3 of the MPI 4.0 standard

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

### CODE REVIEW

**Comment by jjhursey** [comment]
bot:ibm:retest

**Comment by hppritcha** [comment]
@awlauria could you review this when you get a chance?

File: ompi/group/group_init.c [hunk 1]
```diff
@@ -121,12 +121,16 @@ ompi_group_t *ompi_group_allocate_plist_w_procs (ompi_proc_t **procs, int group_
     new_group->grp_my_rank = MPI_UNDEFINED;
     OMPI_GROUP_SET_DENSE(new_group);
 
+    if (NULL != orig_group) {
+        new_group->grp_instance = orig_group->grp_instance;
```
Comment (awlauria) [inline ompi/group/group_init.c:125]: I don't recall if OBJ_NEW will zero out the pointer - should this be initialized to NULL if not set here?

**Review by awlauria — APPROVED** [review]
Just one comment - would be good to null that pointer out.

File: ompi/group/group_init.c (hunk 1 above)
Comment (hppritcha) [reply ompi/group/group_init.c:125]: probably, good catch

**Pull request merged**